### PR TITLE
Tagging

### PR DESCRIPTION
**Pull request by dkonieczek**
With this change, if the repository doesn't contain a tag that matches the version in package.json, a tag with the version will be created and pushed.
If the tag already exists, it will bump or add the suffix to the version and use that as the tag name and push.

https://github.com/searchspring-implementations/snap.searchspring.io/runs/7754442458?check_suite_focus=true

Then we also upload and invalidate under the tag:
https://snapui.searchspring.io/8uyt2m/0.1.0-3/bundle.js

Is this the intended behaviour? In this current state, each action run will be creating a new tag and not updating the version in package.json. Should this only be done when `snapfu patch` is ran instead?

### CODE REVIEW

**Review by korgon — CHANGES_REQUESTED** [review]
The intended behavior is to tag on release only - so perhaps this should only happen on pushes into `production` (similar to logic that doees the uploading to `/siteId/` for release.

Since the tagging only happens on production releases we shouldn't need to have separate lighthouse upload for the tag version (it is same as production branch).

The version in `package.json` will only ever contain the version of the Snap release the project is using (ex: 0.34.4) - it will never have the project build version (ex: 0.34.4-1). These project tag versions (henceforth referred to as "project build version") will only be used in future rollback functionality, where we know the latest bundle built will be the latest project build version matching the version found in `package.json`. This strategy should allow for version rollbacks in `package.json` - where the project build version would just continue bumping the latest tag version of that release version.

**Comment by dkonieczek** [comment]
New run only running after merging PR into production branch
https://github.com/searchspring-implementations/snap.searchspring.io/runs/7770901924?check_suite_focus=true

File: action.yml
Comment (korgon) [inline action.yml:238]: Will this happen on both push and pull request runs?

Comment (dkonieczek) [reply action.yml:238]: Nope, `${{ steps.variables.outputs.branch }}` is only ever `production` on the merge of the PR into production. The PR run contains the branch being merged

Comment (korgon) [reply action.yml:238]: Cool - just wanted to be sure :)

**Comment by korgon** [comment]
Can we do some additional testing to be sure this is working as expected?

**Review by korgon — COMMENTED** [review]
Working well in testing. Approved!
Going to wait until we actually want to release to actually approve :)

File: action.yml
Comment (korgon) [inline action.yml:123]: Attempting to use npx prior to setting up node - this is causing issues.

**Review by korgon — CHANGES_REQUESTED** [review]
@dkonieczek Need to resolve conflicts as well as adjust initial start event.

**Review by korgon — DISMISSED** [review]
Looks good, but still need to wait to release this.

**Review by korgon — APPROVED** [review]
Squash it!

**Pull request merged**
